lib/vmpooler/api/v1.rb

```diff
@@ -28,6 +28,60 @@ def need_token!
       validate_token(backend)
     end
 
+    def alias_deref(hash)
+      newhash = {}
+
+      hash.each do |key, val|
+        if backend.exists('vmpooler__ready__' + key)
+          newhash[key] = val
+        else
+          if Vmpooler::API.settings.config[:alias][key]
+            newkey = Vmpooler::API.settings.config[:alias][key]
+            newhash[newkey] = val
+          end
+        end
+      end
+
+      newhash
+    end
+
+    def checkout_vm(template, result)
```

Comment:
Yeah, the [essentially] same code block was included twice for both `POST /vm` and `POST /vm/:template`, so I wrapped it up into it's own method (`checkout_vm`).